sdb/commands/stacks.py

```diff
@@ -321,56 +374,43 @@ def _call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
                 pass
 
             stack_aggr[tuple(stack_key)].append(task)
+        return stack_aggr
 
-        for stack_key, tasks in sorted(stack_aggr.items(),
-                                       key=lambda x: len(x[1]),
-                                       reverse=True):
-            task_state = stack_key[0]
-            if self.args.tstate and self.args.tstate != task_state:
-                continue
-
-            stacktrace_info = ""
-            if self.args.all:
-                for task in tasks:
-                    stacktrace_info += "{:<18s} {:<16s}\n".format(
-                        hex(task.value_()), task_state)
-            else:
-                stacktrace_info += "{:<18s} {:<16s} {:6d}\n".format(
-                    hex(tasks[0].value_()), task_state, len(tasks))
+    #
+    # Iterate through all objects (task_structs), printing (if specified) and
+    # returning the ones which match any command arguments.
+    #
+    def gather_threads(self, objs: Iterable[drgn.Object],
+                       to_print: bool) -> Iterable[drgn.Object]:
+        #
+        # This implementation only works for linux kernel targets
+        # (crash dumps or live systems). When support for userland is added we can
+        # refactor the kernel code into its own function and switch to the correct
+        # codepath depending on the target.
+        #
+        if not sdb.get_target_flags() & drgn.ProgramFlags.IS_LINUX_KERNEL:
+            raise sdb.CommandError(self.name,
+                                   "userland targets are not supported yet")
+        self.validate_args(self.args)
```

Comment:
This part doesn't have to do anything with "gathering threads", can we wrap it in its own method (`validate_context` maybe?) and call it from both `pretty_print` and `no_input` ?